modules/gatk_rnaseq/1.0/gatk_rnaseq.smk

```diff
@@ -0,0 +1,319 @@
+        stderr = CFG["logs"]["base_recal_report"] + "{seq_type}--{genome_build}--{pair_status}/{sample_id}.gatk_base_recal.stderr.log"
+    params:
+        dbsnp = reference_files("genomes/{genome_build}/variation/dbsnp.common_all-151.vcf.gz"),
+        gnomad = reference_files("genomes/{genome_build}/variation/af-only-gnomad.{genome_build}.vcf.gz"),
+        mem_mb = lambda wildcards, resources: int(resources.mem_mb / 1000)
+    conda:
+        CFG["conda_envs"]["gatk_rnaseq"]
+    threads: CFG["threads"]["gatk_base_recalibration"]
+    resources:
+        **CFG["resources"]["gatk_base_recalibration"]
+    shell:
+        op.as_one_line("""
+        gatk --java-options "-Xmx{params.mem_mb}G" BaseRecalibrator 
+        -R {input.fasta} -I {input.bam} -known-sites {params.dbsnp} -known-sites {params.gnomad} -O {output.table} > {log.stdout} 2> {log.stderr}
+        """)
+        
+rule _gatk_applybqsr:
+    input:
+        bam = rules._gatk_splitntrim.output.bam,
+        table = rules._gatk_base_recalibration.output.table,
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    output:
+        bam = temp(CFG["dirs"]["gatk_applybqsr"] + "{seq_type}--{genome_build}--{pair_status}/{sample_id}.recalibrated.bam"),
+        bai = temp(CFG["dirs"]["gatk_applybqsr"] + "{seq_type}--{genome_build}--{pair_status}/{sample_id}.recalibrated.bai")
+    log:
+        stdout = CFG["logs"]["gatk_applybqsr"] + "{seq_type}--{genome_build}--{pair_status}/{sample_id}.gatk_base_recal.stdout.log",
+        stderr = CFG["logs"]["gatk_applybqsr"] + "{seq_type}--{genome_build}--{pair_status}/{sample_id}.gatk_base_recal.stderr.log"
+    params:
+        mem_mb = lambda wildcards, resources: int(resources.mem_mb / 1000)
+    conda:
+        CFG["conda_envs"]["gatk_rnaseq"]
+    threads: CFG["threads"]["gatk_applybqsr"]
+    resources:
+        **CFG["resources"]["gatk_applybqsr"]
+    shell:
+        op.as_one_line("""
+        gatk --java-options "-Xmx{params.mem_mb}G" ApplyBQSR 
+        -R {input.fasta} -I {input.bam} -bqsr-recal-file {input.table} -O {output.bam} > {log.stdout} 2> {log.stderr}
+        """)
+        
+# Symlink chromosomes used for parallelization
+checkpoint _gatk_rnaseq_input_chrs:
+    input:
+        chrs = reference_files("genomes/{genome_build}/genome_fasta/main_chromosomes.txt")
+    output:
+        chrs = CFG["dirs"]["inputs"] + "chroms/{genome_build}/main_chromosomes.txt"
+    run:
+        op.relative_symlink(input.chrs, output.chrs)
+        
+        
+rule _gatk_variant_calling:
+    input:
+        bam = rules._gatk_applybqsr.output.bam,
+        bai = rules._gatk_applybqsr.output.bai,
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    output:
+        vcf = temp(CFG["dirs"]["gatk_variant_calling"] + "{seq_type}--{genome_build}--{pair_status}/{sample_id}.{chrom}.vcf.gz"),
```

Comment:
Does HaplotypeCaller and VariantFiltration compress vcf files by default, or do you need to do that as a separate step? I was having impression that majority of variant calling tools do not compress by default, but may be very wrong!

Reply:
Yes, the output files are gzipped (tested with: `file results/gatk_rnaseq-1.0/06-gatk_variant_filtration/mrna--grch37--no_normal/Toledo.filtered.vcf.gz`): output:  gzip compressed data, extra field